department-of-veterans-affairs/caseflow
# Adding Delete Modal and delete functionality.

Resolves [APPEALS-45667](https://jira.devops.va.gov/browse/APPEALS-45667)

# Description
Please explain the changes you made here.

- Added Delete Search modal.
- Added Delete button on the view saved searches pages.
- Added Jest test
- 
## Acceptance Criteria

- Create ‘Delete’ button which displays only on My saved search tab of Saved Searches page (not All Saved Searches)
- Enable button after single-select radio button from list of searches is selected
- After button selection, ‘Delete search’ modal displays
- Submission of delete modal sends deletion request to backend
- On successful deletion, present banner confirming deletion 

## Testing Plan

1. Go to [Jira Issue/Test Plan Link](https://jira.devops.va.gov/browse/JIRA-12345) or list them below
2. Login as VHAADMIN 
3. Redirect to /decision_reviews/vha/searches page
4. See Delete button is present at the bottom of the page which should be disabled. 
5. Select one of the radio button in the table, and check if the delete button is enabled. 
6. Click on the delete button and verify a modal pops up [here is the link to figma.](https://www.figma.com/proto/z6aZcfRQldjQRSL3nATeac/Appeals-31802---Saved-Search?page-id=26%3A3449&type=design&node-id=55-15345&viewport=1088%2C425%2C0.04&t=z9c3dqpvKjAKkjby-1&scaling=scale-down&starting-point-node-id=110%3A64866&show-proto-sidebar=1)
7. Modal should have title, description and name of a search you selected. 
8. Click on the delete button, pop up should be removed
9. you should be redirected back to Saved Searches page which should have an alert on the top of the page. 
10. verify the alert message matches the message in the figma design. 


# Frontend
## User Facing Changes

 BEFORE|AFTER
 ---|---

## Storybook Story
*For Frontend (Presentation) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
